Sources/GRPC/LengthPrefixedMessageWriter.swift

```diff
@@ -41,6 +41,63 @@ internal struct LengthPrefixedMessageWriter {
     }
   }
 
+  private func compress(
+    buffer: ByteBuffer,
+    into output: inout ByteBuffer,
+    using compressor: Zlib.Deflate
+  ) throws {
+    // Set the compression byte.
+    output.writeInteger(UInt8(1))
+
+    // Leave a gap for the length, we'll set it in a moment.
+    let payloadSizeIndex = output.writerIndex
+    output.moveWriterIndex(forwardBy: MemoryLayout<UInt32>.size)
+
+    // Compress the message. We know that we need to drop the first 5 bytes, and we know that these
+    // bytes must exist.
+    var buffer = buffer
+    buffer.moveReaderIndex(forwardBy: 5)
+    let bytesWritten = try compressor.deflate(&buffer, into: &output)
```

Comment:
This scrambles `output` if it throws: is that intentional?

Reply:
It is not. Good catch!